store/file/q1q4_file.go

```diff
@@ -0,0 +1,135 @@
+package file
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+	eds "github.com/celestiaorg/celestia-node/share/new_eds"
+	"github.com/celestiaorg/celestia-node/share/shwap"
+)
+
+var _ eds.AccessorCloser = (*Q1Q4File)(nil)
+
+type Q1Q4File struct {
+	ods *ODSFile
+}
+
+func OpenQ1Q4File(path string) (*Q1Q4File, error) {
+	ods, err := OpenODSFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Q1Q4File{
+		ods: ods,
+	}, nil
+}
+
+func CreateQ1Q4File(path string, datahash share.DataHash, eds *rsmt2d.ExtendedDataSquare,
+) (*Q1Q4File, error) {
+	ods, err := CreateODSFile(path, datahash, eds)
+	if err != nil {
+		return nil, err
+	}
+
+	err = writeQ4(ods.fl, eds)
+	if err != nil {
+		return nil, fmt.Errorf("writing Q4: %w", err)
+	}
+
+	return &Q1Q4File{
+		ods: ods,
+	}, nil
+}
+
+func (f *Q1Q4File) Size(ctx context.Context) int {
+	return f.ods.Size(ctx)
+}
+
+func (f *Q1Q4File) Sample(ctx context.Context, rowIdx, colIdx int) (shwap.Sample, error) {
+	// use native AxisHalf implementation, to read axis from Q4 quandrant when possible
+	half, err := f.AxisHalf(ctx, rsmt2d.Row, rowIdx)
+	if err != nil {
+		return shwap.Sample{}, fmt.Errorf("reading axis: %w", err)
+	}
+	shares, err := half.Extended()
+	if err != nil {
+		return shwap.Sample{}, fmt.Errorf("extending shares: %w", err)
+	}
+	return shwap.SampleFromShares(shares, rsmt2d.Row, rowIdx, colIdx)
+}
```

Comment:
My understanding is that Sample won't ever be called if it is wrapped with CacheAccessor. The same is true for RowNamespaceData. This is because cache only calls AxisHalf of the underlying Accessor. Correct?

Reply:
Yes, but cache is decoupled, so Q1Q4 file works in the most optimal way in assumption of no cache. It also helps to have benchmarks of performance of cache-less Q1Q4 file.